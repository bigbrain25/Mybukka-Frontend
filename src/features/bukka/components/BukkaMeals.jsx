import React, { Fragment } from 'react';

import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import Container from 'Components/container';
import Row from 'Components/grid/Row';
import Column from 'Components/grid/Column';

import MealCard from 'Components/card/MealCard';


import './bukkaMeals.scss';
import NoResult from '../../../components/not-found/NoResult';

const BukkaMealsHeader = ({ category }) => (
  <div className="bukka-meals-header" id={category}>
    <h4 className="header-text">{category}</h4>
  </div>
);

const BukkaMeals = ({ bukkaMenu, searchQuery, isInSearch, hasNoResult }) => {
  const bukkaCategories = [
    ...new Set(bukkaMenu.map(mealData => mealData.category))
  ];

  if (bukkaMenu.length <= 0
    || Object.keys(bukkaMenu[0]).length <= 0) {
    return null;
  }

  const onSearch = (category, items) => {
    const isCategory = item => item.category === category;
    return Array.isArray(items) ? items.filter(menu => isCategory(menu) && isInSearch(menu))
      : isCategory(items) && isInSearch(items);
  };

  return (
    <Container classNames={hasNoResult() ? '' : 'menu-catalogs'}>
      {bukkaCategories.map(eachCategory => (
        onSearch(eachCategory, bukkaMenu).length > 0 &&
        <Fragment key={`store-menu-catelogs-${eachCategory}`}>
          <BukkaMealsHeader category={eachCategory} />
          <Row classNames="menu-section">
            {bukkaMenu.map(mealData => (
              <Fragment key={`store-menu-catelogs-${mealData.title}-${mealData._id}`}>
                {onSearch(eachCategory, mealData) && (
                  <Column classNames="col-12 col-lg-6 col-xl-6 col-xs-12 col-sm-12 meal-column">
                    <MealCard {...mealData} />
                  </Column>
                )}
              </Fragment>
            ))}
          </Row>
        </Fragment>
      ))}
      {hasNoResult() && <NoResult withPadding text={searchQuery} />}
    </Container>
  );
};

const mapStateToProps = ({ productsReducer: { bukkaMenu } }) => ({
  bukkaMenu
});

export default connect(
  mapStateToProps,
)(BukkaMeals);

BukkaMealsHeader.propTypes = {
  category: PropTypes.string.isRequired
};

BukkaMeals.propTypes = {
  bukkaMenu: PropTypes.arrayOf(
    PropTypes.shape({
      category: PropTypes.string,
      description: PropTypes.string,
      price: PropTypes.number,
      title: PropTypes.string,
      name: PropTypes.string,
      imageUrl: PropTypes.string
    })
  ).isRequired,
  fetchBukkaMenu: PropTypes.func.isRequired,
};
