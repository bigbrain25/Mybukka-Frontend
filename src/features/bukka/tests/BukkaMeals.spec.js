import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import BukkaMeals from '../components/BukkaMeals';

const initialState = {
  navbarAuthReducer: { type: 'Sign In', },
  deliveryModeReducer: { mode: 'delivery', },
  checkoutModeReducer: { mode: true },
  locationsPredictionReducer: { predictions: [3.7474, 3.4848] },
  authenticationReducer: {
    status: {
      authenticated: false,
      error: false
    }
  },
  fetchBukkaMenuReducer: { bukkaMenu: [
    {
      title: 'title',
      imageUrl: 'www.imageUrl.com',
      description: 'your description',
      price: 4454,
      category: 'african',
      slug: 'any-way-slug'
    }
  ] }
};

const store = mockStore(initialState);

describe('BukkaMeals component', () => {
  const props = {
    mealsData: [
      {
        name: 'sample meal',
        category: 'Breakfast',
        title: 'Amala',
        description:
          'A very great plate of Amala that will make you beg for more',
        imageUrl:
          'https://res.cloudinary.com/dn93xk5ni/image/upload/v1549933392/burrito-chicken-close-up-461198_g9llka.jpg', // eslint-disable-line
        price: 30
      },
      {
        name: 'sample meal',
        category: 'Breakfast',
        title: 'Amala',
        description:
          'A very great plate of Amala that will make you beg for more',
        imageUrl:
          'https://res.cloudinary.com/dn93xk5ni/image/upload/v1549933392/burrito-chicken-close-up-461198_g9llka.jpg', // eslint-disable-line
        price: 30
      }
    ]
  };

  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <BukkaMeals {...props} />
      </MemoryRouter>
    </Provider>
  );

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
