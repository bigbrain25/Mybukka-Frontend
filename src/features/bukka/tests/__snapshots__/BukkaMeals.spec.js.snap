// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BukkaMeals component renders properly 1`] = `ReactWrapper {}`;
